### ISSUE TITLE
Cop to suggest using in place functions (e.g. sort!) when shorter

### ISSUE THREAD

**Issue by madwort** [rubocop-hq/rubocop#6463]
**Is your feature request related to a problem? Please describe.**

The legacy codebase I'm working with features examples of the pattern `myvar = myvar.sort { |a, b| a<=>b }` and I would like to use rubocop to automate finding these.

**Describe the solution you'd like**

I would like a cop that will suggest the shorter `myvar.sort! { |a, b| a <=> b }` in place of the above example. 

**Describe alternatives you've considered**

I have not considered any alternatives.

**Additional context**

**Comment by kurko** [comment]
I'm not sure this should be done.

```rb
irb(main):003:0> var1 = [1, 2, 0]
=> [1, 2, 0]
irb(main):004:0> def mysort(h); h.sort!; end
=> :mysort
irb(main):005:0> mysort(var1)
=> [0, 1, 2]
irb(main):006:0> var1 # global var1 got mutated
=> [0, 1, 2]
irb(main):007:0> var2 = [1, 2, 0]
=> [1, 2, 0]
irb(main):008:0> def mysort(h); h = h.sort; h; end
=> :mysort
irb(main):009:0> mysort(var2)
=> [0, 1, 2]
irb(main):010:0> var2 # global var2 wasn't mutated
=> [1, 2, 0]
```

So `x = x.sort` and `x.sort!` are not the same thing, even if they're encapsulated within a function (the reference is used and it will mutate things outside the function).

**Comment by madwort** [comment]
Good point @kurko , closing this suggestion as an unsafe suggestion is not very useful.
